src/engine/profiler.cc

```diff
@@ -16,8 +16,7 @@
 namespace mxnet {
 namespace engine {
 
-Profiler* Profiler::instance_ = nullptr;
-std::mutex Profiler::m_;
+Profiler* Profiler::instance_ = new Profiler();
```

Comment:
For current implementation, I think this is fine. But please be aware that this means the profiler object will always be created no matter it is used or not.